dynawo/sources/Modeler/DataInterface/test/TestDataInterface.cpp

```diff
@@ -0,0 +1,292 @@
+  parametersSet->createParameter("load_alpha", 0.);
+  parametersSet->createParameter("load_beta", 0.);
+  parametersSet->createParameter("load_isRestorative", false);
+  parametersSet->createParameter("load_isControllable", false);
+  parametersSet->createParameter("transformer_t1st_THT", 9.);
+  parametersSet->createParameter("transformer_tNext_THT", 10.);
+  parametersSet->createParameter("transformer_t1st_HT", 11.);
+  parametersSet->createParameter("transformer_tNext_HT", 12.);
+  parametersSet->createParameter("transformer_tolV", 13.);
+  modelNetwork->setPARParameters(parametersSet);
+
+  return modelNetwork;
+}  // initializeModel(shared_ptr<DataInterface> data);
+
+void
+exportStateVariables(shared_ptr<DataInterface> data) {
+  shared_ptr<SubModel> modelNetwork = initializeModel(data);
+  ModelMulti mm;
+  mm.addSubModel(modelNetwork, "MyLib");
+  mm.initBuffers();
+  mm.init(0.0);
+  data->getStateVariableReference();
+  data->exportStateVariables();
+  data->updateFromModel(false);
+  data->importStaticParameters();
+}
+
+TEST(DataInterfaceTest, testLostEquipments) {
+#ifdef USE_POWSYBL
+  auto network = boost::make_shared<powsybl::iidm::Network>("MyNetwork", "MyNetwork");
+
+  auto& substation = network->newSubstation()
+    .setId("MySubStation")
+    .add();
+
+  auto& vl = substation.newVoltageLevel()
+    .setId("MyVoltageLevel")
+    .setNominalV(5.)
+    .setTopologyKind(powsybl::iidm::TopologyKind::BUS_BREAKER)
+    .add();
+
+  vl.getBusBreakerView().newBus().setId("MyBus1").add();
+  vl.getBusBreakerView().newBus().setId("MyBus2").add();
+
+  vl.getBusBreakerView().newSwitch()
+    .setId("MySwitch")
+    .setBus1("MyBus1")
+    .setBus2("MyBus2")
+    .add();
+
+  vl.newLoad()
+    .setId("MyLoad")
+    .setBus("MyBus1")
+    .setP0(50.0)
+    .setQ0(40.0)
+    .add();
+
+  vl.newGenerator()
+    .setId("MyGenerator")
+    .setBus("MyBus2")
+    .setMaxP(50.0)
+    .setMinP(3.0)
+    .setTargetP(45.0)
+    .setTargetQ(5.0)
+    .setVoltageRegulatorOn(false)
+    .add();
+#else
+  IIDM::connection_status_t cs = {true /*connected*/};
+  IIDM::Port p1("MyBus1", cs), p2("MyBus2", cs);
+  IIDM::Connection c1("MyVoltageLevel", p1, IIDM::side_1), c2("MyVoltageLevel", p2, IIDM::side_2);
+
+  IIDM::builders::NetworkBuilder nb;
+  boost::shared_ptr<IIDM::Network> network = boost::make_shared<IIDM::Network>(nb.build("MyNetwork"));
+
+  IIDM::builders::SubstationBuilder ssb;
+  IIDM::Substation ss = ssb.build("MySubStation");
+
+  IIDM::builders::VoltageLevelBuilder vlb;
+  vlb.mode(IIDM::VoltageLevel::bus_breaker);
+  vlb.nominalV(5.);
+  IIDM::VoltageLevel vl = vlb.build("MyVoltageLevel");
+
+  IIDM::builders::BusBuilder bb;
+  vl.add(bb.build("MyBus1"));
+  vl.add(bb.build("MyBus2"));
+
+  IIDM::builders::SwitchBuilder sb;
+  vl.add(sb.build("MySwitch"), "MyBus1", "MyBus2");
+
+  IIDM::builders::LoadBuilder lb;
+  lb.p0(50.);
+  lb.q0(40.);
+  vl.add(lb.build("MyLoad"), c1);
+
```

Comment:
Please add one test in which two components are lost equipments

Reply:
Test added with three lost equipments